ddl/table.go

```diff
@@ -42,6 +43,7 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 	tbInfo.State = model.StateNone
 	err := checkTableNotExists(t, job, schemaID, tbInfo.Name.L)
 	if err != nil {
+		job.State = model.JobStateCancelled
```

Comment:
set to model.JobStateCancelled is done in checkTableNotExists when table exists, other error we should retry, so don't set to canceled.